.azuredevops/pipelineTemplates/module.jobs.deploy.yml

```diff
@@ -0,0 +1,287 @@
+
+                $templateFilePath = ([String]::IsNullOrEmpty('${{ deploymentBlock.templateFilePath }}')) ?
+                  (Join-Path '$(ModuleRepoRoot)' '$(modulePath)' 'deploy.bicep') :
+                  (Join-Path '$(moduleRepoRoot)' '${{ deploymentBlock.templateFilePath }}')
+
+                # ----------------- #
+                # INVOKE DEPLOYMENT #
+                # ----------------- #
+                $functionInput = @{
+                  templateFilePath   = $templateFilePath
+                  parameterFilePath  = Join-Path '$(parametersRepoRoot)' '${{ deploymentBlock.path }}'
+                  location           = '${{ parameters.location }}'
+                  resourceGroupName  = '${{ parameters.resourceGroupName }}'
+                  subscriptionId     = '${{ parameters.subscriptionId }}'
+                  managementGroupId  = '${{ parameters.managementGroupId }}'
+                  doNotThrow         = $true
+                }
+
+                Write-Verbose "Invoke task with" -Verbose
+                Write-Verbose ($functionInput | ConvertTo-Json | Out-String) -Verbose
+
+                # Invoke deployment
+                $res = New-ModuleDeployment @functionInput -Verbose
+
+                # Get deployment name
+                Write-Host ('##vso[task.setvariable variable=deploymentName]{0}' -f $res.deploymentName)
+
+                # Populate further outputs
+                foreach ($outputKey in $res.deploymentOutput.Keys) {
+                    Write-Output ('##vso[task.setvariable variable={0}]{1}' -f $outputKey, $res.deploymentOutput[$outputKey].Value)
+                }
+
+                if ($res.ContainsKey('exception')) {
+                  # Happens only if there is an exception
+                  throw $res.exception
+                }
+
+          # [Removal] task(s)
+          #------------------
+          - task: AzurePowerShell@5
+            displayName: 'Remove deployed resources via [${{ parameters.serviceConnection }}]'
+            condition: and(eq( '${{ parameters.removeDeployment }}', 'true'), not(eq(variables['deploymentName'],'')))
+            inputs:
+              azureSubscription: ${{ parameters.serviceConnection }}
+              azurePowerShellVersion: ${{ parameters.azurePowerShellVersion }}
+              preferredAzurePowerShellVersion: ${{ parameters.preferredAzurePowerShellVersion }}
+              ScriptType: InlineScript
+              failOnStandardError: false
+              pwsh: true
+              inline: |
+                # Load used function
+                . (Join-Path '$(moduleRepoRoot)' '$(pipelineFunctionsPath)' 'resourceRemoval' 'Remove-DeployedModule.ps1')
+
+                $templateFilePath = ([String]::IsNullOrEmpty('${{ deploymentBlock.templateFilePath }}')) ?
+                  (Join-Path '$(ModuleRepoRoot)' '$(modulePath)' 'deploy.bicep') :
```

Comment:
Can we make this more interoperable with json too?

Reply:
Good point. Will cover that. Damn

Reply:
done